src/test/java/dev/ai4j/openai4j/chat/RequestDisabledChatCompletionAsyncTest.java

```diff
@@ -0,0 +1,280 @@
+package dev.ai4j.openai4j.chat;
+
+import static dev.ai4j.openai4j.chat.ChatCompletionModel.GPT_4_VISION_PREVIEW;
+import static dev.ai4j.openai4j.chat.ChatCompletionTest.*;
+import static dev.ai4j.openai4j.chat.ResponseFormatType.*;
+import static java.util.Collections.singletonMap;
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static org.assertj.core.api.Assertions.*;
+import static org.junit.jupiter.params.provider.EnumSource.Mode.*;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import dev.ai4j.openai4j.OpenAiClient;
+import dev.ai4j.openai4j.OpenAiHttpException;
+
+class RequestDisabledChatCompletionAsyncTest {
+
+    private final OpenAiClient client = OpenAiClient.builder()
+            .disableRequests()
+            .logRequests()
+            .logResponses()
+            .build();
+
+    @Test
+    void testSimpleApi() {
+
+        // when
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        client.chatCompletion(USER_MESSAGE)
+                .onResponse(future::complete)
+                .onError(future::completeExceptionally)
+                .execute();
+
+        assertThat(future.isCompletedExceptionally()).isTrue();
+        assertThatExceptionOfType(ExecutionException.class)
+            .isThrownBy(() -> future.get(30, SECONDS))
+            .havingCause()
+            .isInstanceOf(OpenAiHttpException.class)
+            .withMessage("Requests to the live system are disabled");
+    }
+
+    @ParameterizedTest
+    @EnumSource(value = ChatCompletionModel.class, mode = EXCLUDE, names = {
+            "GPT_4_32K", "GPT_4_32K_0314", "GPT_4_32K_0613", // I don't have access to these models
+            "GPT_4_VISION_PREVIEW" // Does not support many things now, including logit_bias and response_format
+    })
+    void testCustomizableApi(ChatCompletionModel model) {
+
+        // given
+        ChatCompletionRequest request = ChatCompletionRequest.builder()
+                .model(model)
+                .addSystemMessage(SYSTEM_MESSAGE)
+                .addUserMessage(USER_MESSAGE)
+                .temperature(1.0)
+                .topP(0.1)
+                .n(1)
+                .stream(false)
+                .stop("one", "two")
+                .maxTokens(3)
+                .presencePenalty(0.0)
+                .frequencyPenalty(0.0)
+                .logitBias(singletonMap("50256", -100))
+                .user("Klaus")
+                .responseFormat(TEXT)
+                .seed(42)
+                .build();
```

Comment:
it is not disabled?

Reply:
It is. The enabling/disabling is on the client itself, not on the `ChatCompletionRequest`. I think thats what we are discussing in the comments though, whether that is the right approach.

Reply:
my bad, I thought it is a model configuration